app/services/borrow_direct_reshare_client.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require 'http'
+
+class BorrowDirectReshareClient
+  DEFAULT_HEADERS = {
+    accept: 'application/json, text/plain',
+    content_type: 'application/json'
+  }.freeze
+
+  attr_reader :base_url
+
+  def initialize(url: Settings.borrow_direct_reshare.url,
+                 username: Settings.borrow_direct_reshare.username,
+                 password: Settings.borrow_direct_reshare.password,
+                 tenant: Settings.borrow_direct_reshare.tenant)
+    uri = URI.parse(url)
+
+    @base_url = url
+    @username = username
+    @password = password
+
+    if uri.user
+      @username ||= uri.user
+      @password ||= uri.password
+      @base_url = uri.dup.tap do |u|
+        u.user = nil
+        u.password = nil
+      end.to_s
+    end
+
+    @tenant = tenant
+  end
+
+  def get(path, **kwargs)
+    authenticated_request(path, method: :get, **kwargs)
+  end
+
+  def get_json(path, **kwargs)
+    parse(get(path, **kwargs))
+  end
+
+  def requests(university_id)
```

Comment:
Why do we need to handle the case where there is no university_id?  Should this be an ArgumentError?

Reply:
I think you're right that this shouldn't be handled here. I'll add a guard clause on BorrowDirectReshareRequests#reshare_requests to handle the case where a patron doesn't have a university_id.